src/test/java/com/butbetter/applicationServices/CalculatorTest_localhost.java

```diff
@@ -0,0 +1,51 @@
+package com.butbetter.applicationServices;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.ResourceAccessException;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@SpringBootTest
+class CalculatorTest_localhost {
+
+    @Autowired
+    private Calculator calculator;
+
+    //integration Testing on localhost:
+
+    //these tests assert to true if the localhost calc api is running
+    @Test
+    @DisplayName("test calc API with the values 100 10")
+    void testCalcApiReturnValueFor10percentFrom100(){
+        calculator.setCalcApiUrlStart("http://localhost:8080/calc/v1/VAT?price=");
+        assertEquals(110,calculator.calculateVATofPrice(100,10));
+    }
+
+    @Test
+    @DisplayName("test calc API with the values -100 10")
+    void testCalcApiReturnForIllegalInputPrice(){
+        calculator.setCalcApiUrlStart("http://localhost:8080/calc/v1/VAT?price=");
+        assertThrows(HttpClientErrorException.BadRequest.class, ()-> calculator.calculateVATofPrice(-100,10));
+    }
+
+    @Test
+    @DisplayName("test calc API with the values 100 -10")
+    void testCalcApiReturnForIllegalInputPercent(){
+        calculator.setCalcApiUrlStart("http://localhost:8080/calc/v1/VAT?price=");
+        assertThrows(HttpClientErrorException.BadRequest.class, ()-> calculator.calculateVATofPrice(100,-10));
```

Comment:
Why are you testing the calu api ??? The Api is blackbox for you

Reply:
This Api have the same tests in calculate api

Reply:
im testing the calculator service in combination with the calc Rest API because mocking a server is tricky
i also created a test class mocking the calc Rest API but, by testing in combination i was able to cover more test cases.
This is not a Unit test obviously, but more integration/system test